File: sorting/selection-sort/selection-sort.py

list = [7, 5, 8, 12, 2, 9, 10];

def selection_sorting(list):
 for i in range(len(list)-1):
  min_index = i
  for j in range(i+1,len(list)):
   if list[min_index] > list[j]:
     min_index = j

  list[min_index] ,list[i] = list[i],list[min_index]
  return list
 

selection_sorting(list)

# new arr [ 2, 5, 7, 8, 9, 10, 12 ]

print(selection_sorting(list))

File: sorting/selection-sort/selection-sort.js
const arr = [7, 5, 8, 12, 2, 9, 10];

function selectionSorting(arr) {
  for (let i = 0; i < arr.length; i++) {
    let minIndex = i;
    for (let j = i + 1; j < arr.length; j++) {
      if (arr[minIndex] > arr[j]) {
        minIndex = j;
      }
    }
    let temp = arr[i];
    arr[i] = arr[minIndex];
    arr[minIndex] = temp;
  }

  return arr;
}

selectionSorting(arr);

console.log(selectionSorting(arr));
